spec/features/course/homepage_spec.rb

```diff
@@ -6,10 +6,23 @@
 
   with_tenant(:instance) do
     let(:course) { create(:course, :opened) }
-    let(:achievement_feed_notification) do
-      achievement = create(:course_achievement)
+    let(:feed_notifications) do
```

Comment:
I think fundamentally this is something that people would have to grapple with. The Object-relational pattern requires the coupling of models by definition (joins) and thus these dependencies are necessary. We should be optimising for the common case, and not the edge case of new people joining. I'm not saying make things complex, I'm saying where there is complexity, optimise for the most common case.